Parser/tokenizer.c

```diff
@@ -1793,8 +1793,14 @@ tok_get_normal_mode(struct tok_state *tok, tokenizer_mode* current_tok, struct t
     /* Skip comment, unless it's a type comment */
     if (c == '#') {
 
-        if (INSIDE_FSTRING(tok)) {
-            return MAKE_TOKEN(syntaxerror(tok, "f-string expression part cannot include '#'"));
+        // We don't allow # inside a one-line f-string expr,
+        // but we do in multi-line f-string expr
+        if (INSIDE_FSTRING(tok) && (current_tok->f_string_quote_size == 1)) {
+            while (c != '\n') {
+                c = tok_nextc(tok);
+                if (c == '}')
```

Comment:
This may raise "false positives". What if for instance the text contains sone dictionary literal or escaped }} or anything similar?

Reply:
I tried to play around with more cases, but I couldn't find a false-positive:
```py
>>> f"# something"
'# something'
>>> f"# {1}"
'# 1'
>>> f"{{}} # {{}}"
'{} # {}'
>>> f"# {{f'# {{}}'}}"
"# {f'# {}'}"
>>> f"""{f"{f'''{f'# {1} #'}'''}"}"""
'# 1 #'
>>> d = {"a": 1}
>>> f"{''.join(f" # {k}: {v}" for k, v in d.items())}"
' # a: 1'
```
Maybe you have something else in mind? :thinking: 
Perhaps you were thinking of using  `INSIDE_FSTRING_EXPR(current_tok)` as well?

Reply:
Another example:

```
>>> f"{sadasdfsd # {1,2,3,4}

    }"
  File "/Users/pgalindo3/github/python/main/lel.py", line 1
    f"{sadasdfsd # {1,2,3,4}
                           ^
SyntaxError: single line f-string expression part cannot include '#'

```

Reply:
I think we may just want to drop this check because IIRC the error you will get is that there is a { that is never closed and it should point correctly to the one that's not closed

Reply:
yeah, also then we would need to define if we close the expression the moment we find a '}' or we agreed on something else, because something like:
```
f"The result is: {2 # a nice number} and {17 # a prime number}"
```
should work by that condition, but things like:
```
f"something : {2 # I like the '}' parenthesis}"
#                                            ^ this is the one that closes the expr
```
will not. And the only solution I think could be possible, would be to either allow the '}}' escape pattern, so it could be valid with:
```
f"something : {2 # I like the '}}' parenthesis}"
```
or we simply don't allow having `{` or `}` in the comments while being inside a f-string expression.

Not sure if we have another workaround.

Reply:
Nah I think the approach here is that a comment is a comment and we should not try to be clever about this. Let allow any comment inside and let's try to focus on the fact that the error message makes sort of sense (I expect that most of them would be unclosed brackets).